src/Vault.sol

```diff
@@ -234,45 +236,41 @@ contract Vault {
 
     /**
-     * @notice Can be called by authorised applications to open or increase a margin position.
+     * @notice Can be called by authorised applications to increase a margin position.
      * @param baseCurrency The Base-currency in which the margin position is denominated
      * @param amount The amount the position is increased.
      * @return success Boolean indicating if there is sufficient free margin to increase the margin position
-     * @dev All values expressed in the base currency of the vault with same number of decimals as the base currency.
-     * @dev Since increasing margin position is financial activity, liquidation threshold update is done here.
      */
     function increaseMarginPosition(address baseCurrency, uint256 amount)
         public
         onlyAuthorized
         returns (bool success)
     {
         if (baseCurrency != vault.baseCurrency) {
-            _setBaseCurrency(baseCurrency);
+            return false;
         }
-        success = getFreeMargin() >= amount;
-        // Update the vault values
         (address[] memory assetAddresses, uint256[] memory assetIds, uint256[] memory assetAmounts) =
             generateAssetData();
-        vault.liqThres =
-            IRegistry(registry).getLiquidationThreshold(assetAddresses, assetIds, assetAmounts, vault.baseCurrency);
+        (uint256 collateralValue, uint256 liquidationThreshold) = IRegistry(registry)
+            .getCollateralValueAndLiquidationThreshold(assetAddresses, assetIds, assetAmounts, vault.baseCurrency);
+
+        // Check that the collateral value is bigger than the sum  of the already used margin and the increase
+        // ToDo: For trusted protocols, already pass usedMargin with the call -> avoid additional hop back to trusted protocol to fetch already open debt
+        success = collateralValue >= getUsedMargin() + amount;
+
+        // Can safely cast to uint16 since liquidationThreshold is maximal 10000
+        if (success) vault.liqThres = uint16(liquidationThreshold);
     }
 
     /**
      * @notice Can be called by authorised applications to close or decrease a margin position.
      * @param baseCurrency The Base-currency in which the margin position is denominated.
      * @dev All values expressed in the base currency of the vault with same number of decimals as the base currency.
      * @return success Boolean indicating if there the margin position is successfully decreased.
-     * @dev Since decreasing margin position is financial activity, liquidation threshold update is done here.
      * @dev ToDo: Function mainly necessary for integration with untrusted protocols, which is not yet implemnted.
      */
-    function decreaseMarginPosition(address baseCurrency, uint256) public onlyAuthorized returns (bool success) {
+    function decreaseMarginPosition(address baseCurrency, uint256) public view onlyAuthorized returns (bool success) {
```

Comment:
should we also update vault.liqThres here? Otherwise you could have a vault with no exposure but still a stored liqthres of X, even though it cannot be liquidated at value X (given that it's recalculated upon liquidation)

Reply:
vault.liqThres doesn't seem updated on liquidation, so it has to be updated here

Reply:
And vaults without cannot be liquidated, even if there is still a liqThresh stored
![image](https://user-images.githubusercontent.com/43761140/210323092-c10197a5-8083-4121-82eb-c9d98a719fa0.png)